File: 151001/Aleksandrov/RESTLab1/discussion/src/main/java/by/bsuir/entities/Comment.java
package by.bsuir.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

import static org.springframework.data.cassandra.core.cql.PrimaryKeyType.CLUSTERED;
import static org.springframework.data.cassandra.core.cql.PrimaryKeyType.PARTITIONED;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(value = "tbl_comment")
public class Comment {

    @PrimaryKeyColumn(name = "country", type = PARTITIONED)
    private String country;

    @PrimaryKeyColumn(name = "issueid", ordinal = 0, type = CLUSTERED)
    private Long issueId;

    @PrimaryKeyColumn(name = "id", ordinal = 1, type = CLUSTERED)
    private Long id;

    private String content;

}

File: 151003/Barylko/NoSQL/Module.Discussion/Models/Comment.cs
using Cassandra.Mapping.Attributes;
namespace Discussion.Models;

[Table("tbl_comment", AllowFiltering = true)]
public class Comment
{
    [PartitionKey] public string Country { get; set; } = string.Empty;

    [ClusteringKey(0)] public long IssueId { get; set; }
    
    [ClusteringKey(1)] public long Id { get; set; }
    
    public string Content { get; set; } = string.Empty;
}
